pkg/registry/experimental/controller/etcd/etcd.go

```diff
@@ -89,6 +91,21 @@ func (r *ScaleREST) Update(ctx api.Context, obj runtime.Object) (runtime.Object,
 	if !ok {
 		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
 	}
+
```

Comment:
I'm against faking old scale object here. It bring no value. I think it is better to just call ValidateScale here on the new object.